### ISSUE TITLE
System Monitors - Add missing Virtualmin services

### ISSUE THREAD

**Issue by shoulders** [webmin/webmin#2236]
|  SYSTEM INFORMATION||
|----------------------|---------------------------|
|  OS type and version |     Ubuntu Linux 22.04.4     |
|  Webmin version      |     2.111      |
|  Usermin version     |     2.010     |
|  Virtualmin version  |     7.20.2       |
|  Theme version       |     21.10      |
|  Package updates     |     All installed packages are up to date   |

### background

in virtualmin there is a tab called `Server Status` that displays Virtualmin related services

![352525665-002e4353-b807-4b68-aecc-2f6f51088510](https://github.com/user-attachments/assets/f5e3ada5-70c1-4116-8af7-adf11c2b2573)

### the issue

`System Monitors` does not have all of these services predefined allowing for easy configuration of their associated monitors

NB: i know there is the ability to add custom monitors that would do the same, but I think the virtualmin ones should be there to make things easier.

### proposed solution

Add the following missing services in to `System Monitor` as predefined options

- PHP-FPM  (you would add a monitor but select the PHP version number)
- Cyrus SASL Authentication server
- Postgrey - (I will ask on Virtualmin repo for this to be added to `Servers Status`)
- FirewallD - (I will ask on Virtualmin repo for this to be added to `Servers Status`)
- fail2ban - (I will ask on Virtualmin repo for this to be added to `Servers Status`)
- ClamAV Virus Scanning Server
- ProFTPD

### related issue

- https://github.com/virtualmin/virtualmin-gpl/issues/870

**Comment by jcameron** [comment]
I agree it would be good to add other services ... I am working on this now.

**Comment by iliajie** [comment]
Excellent! Thank you, Jamie!

**Comment by jcameron** [comment]
By the way, you can already add any server managed by Systemd with the "Bootup Action" monitor type, which covers all those PHP FPM versions, ClamD, FirewallD, etc..

**Comment by iliajie** [comment]
Yes, sure! This is awesome! Though, I think it's much more complex for users, instead of just opening a dropdown and entering the service name, I'd suggest we add the most basic ones, maybe?

**Comment by shoulders** [comment]
@jcameron 

> By the way, you can already add any server managed by Systemd with the "Bootup Action" monitor type, which covers all those PHP FPM versions, ClamD, FirewallD, etc..

Thanks for this info.

This `issue` is about making the experience easy for the virtualmin user and be in sync with Virtualmin  😄 . I appreciate a lot of people who use Webmin are far better at server stuff than I am.

I personally would not have a clue on how to add a monitor as described above but I can edit and enable one easily.

@iliajie I think the additional ones in my list above cover all services in virtualmin and these are the ones I would class as basic. Again this is about making Webmin sync up with the Virtualmin experience. 😄

**Comment by fabien79** [comment]
@shoulders can you just share how you did to setup the php-fpm server with all versions on the server status please ?
Thanks a lot

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
